sdk/core/System.ClientModel/tests/Options/PipelineMessageClassifierTests.cs

```diff
@@ -11,6 +11,82 @@ namespace System.ClientModel.Tests.Options;
 
 public class PipelineMessageClassifierTests
 {
+    [Theory]
+    [TestCase(500)]
+    [TestCase(429)]
+    [TestCase(408)]
+    [TestCase(502)]
+    [TestCase(503)]
+    [TestCase(504)]
+    public void RetriesStatusCodes(int code)
+    {
+        MockPipelineMessage message = new MockPipelineMessage();
+        message.SetResponse(new MockPipelineResponse(code));
```

Comment:
To keep these as pure unit test, could we eliminate `MockPipelineMessage` from this test and operate directly with `PipelineMessageClassifier`? If the implementation of `MockPipelineMessage` or `PipelineMessage` ever changes, this test may end up inadvertently testing some other implementation.

Reply:
That is fair.  I copied these over from Azure.Core, but I also wondered whether it was needed to use the message's classifier here. I'll update to make this consistent with other tests in this file, and reduce the code coverage area to just the classifier.  Thanks!